### DISCUSSION TITLE
YouTube style chapters timeline in video.js

### DISCUSSION THREAD

**Discussion by petarsubotic** [videojs/video.js#8235] [Q&A]
![image](https://user-images.githubusercontent.com/3495684/231184515-873d31be-13d1-4dad-bba8-f9ead604bbbd.png)

We currently use chapter "markers" as points on the timeline and are considering developing timeline sections as in YouTube player.
Has anyone done anything similar in video.js and if so can you help by pointing us to any helpful resources or ideas.

**Comment by getnot** [comment]
Any success in this ?

**Reply by amtins** [reply]
@getnot I made a component few month ago, but I'm still not totally happy with the result, so I didn't publish it (yet?), here is a little demo
[Screencast from 07. 07. 23 22:00:06.webm](https://github.com/videojs/video.js/assets/34163393/2197ee30-add3-4c9d-91e9-569b0a6fbdf8)

I was also playing with another type of [markers](https://github.com/videojs/video.js/issues/8315#issuecomment-1586249226), which I think could be easily updated to have something youtube like
[Screencast from 07. 07. 23 21:55:36.webm](https://github.com/videojs/video.js/assets/34163393/0affa14e-6ef9-4066-a311-e932f251ca8d)

**Comment by petarsubotic** [comment]
in the meantime we made a lot of progress on this, demo:

https://github.com/videojs/video.js/assets/3495684/8706e483-c915-495c-9d95-752368fc18fe

here's the basic coding idea:
https://jsfiddle.net/petarsubotic/qyz94acj/

**Reply by avengsum** [reply]
@petarsubotic Hey how you create this can you give a high-level idea to build this?

**Reply by petarsubotic** [reply]
@avengsum we used a [clipath ](https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path)(see [jsfiddle above](https://jsfiddle.net/petarsubotic/qyz94acj/)) to create a "mask" layer above the regular progress bar and create sections. Additionally we are adjusting the height of the specific path to get a "hover" effect.

**Reply by pstashp** [reply]
@avengsum, Odysee uses the same method too.
- https://odysee.com/@WebDevSimplified:8/how-to-create-3-levels-of-react-image:7
- https://github.com/OdyseeTeam/odysee-frontend/blob/feaa3ffa85484a659419ac16d1e7caca24ce1e1e/ui/component/viewers/videoViewer/internal/chapters.js#L155

They chose not to touch the existing videojs components and only work on top of it. If you plan to expand with more features, it might be easier to just rewrite the whole `ProgressControl` cluster.
